File: VineyardSite/VineyardSite/Model/CartItem.cs
using System.Text.Json.Serialization;

namespace VineyardSite.Model;

public class CartItem
{
    public int Id { get; set; }
    public int CartId { get; set; }
    [JsonIgnore]
    public Cart Cart { get; set; }
    public int WineVariantId { get; set; }
    public WineVariant WineVersion { get; set; }
    public int Quantity { get; set; }
}

File: VineyardSite/AngularFrontend/VineyardSiteAngular/src/app/Models/CartItem.ts
import { ICart } from "./Cart";
import { IWineVariant } from "./WineVariant";

export interface ICartItem {
    Id: number,
    CartId: number,
    Cart: ICart,
    WineVariantId: number,
    WineVersion: IWineVariant,
    Quantity: number
}
